multicluster/controllers/multicluster/commonarea/labelidentityimport_controller_test.go

```diff
@@ -0,0 +1,150 @@
+				ID:    uint32(2),
+			},
+		},
+	}
+
+	labelIdentity1 = &mcsv1alpha1.LabelIdentity{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: resImpNamespacedName.Name,
+		},
+		Spec: mcsv1alpha1.LabelIdentitySpec{
+			Label: "ns:kubernetes.io/metadata.name=ns&pod:app=client",
+			ID:    uint32(1),
+		},
+	}
+
+	labelIdentity2 = &mcsv1alpha1.LabelIdentity{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: resImpNamespacedName.Name,
+		},
+		Spec: mcsv1alpha1.LabelIdentitySpec{
+			Label: "ns:kubernetes.io/metadata.name=ns&pod:app=client",
+			ID:    uint32(2),
+		},
+	}
+)
+
+func TestLabelIdentityResourceImportReconclie(t *testing.T) {
+	tests := []struct {
+		name                    string
+		existLabelIdentity      *mcsv1alpha1.LabelIdentity
+		existResImp             *mcsv1alpha1.ResourceImport
+		resImportNamespacedName types.NamespacedName
+		expLabelIdentity        *mcsv1alpha1.LabelIdentitySpec
+		isDeleted               bool
+	}{
+		{
+			"create LabelIdentity",
+			&mcsv1alpha1.LabelIdentity{},
+			labelIdentityResImp1,
+			resImpNamespacedName,
+			&labelIdentity1.Spec,
+			false,
+		},
+		{
+			"update LabelIdentity",
+			labelIdentity1,
+			labelIdentityResImp2,
+			resImpNamespacedName,
+			&labelIdentity2.Spec,
+			false,
+		},
+		{
+			"delete LabelIdentity",
+			labelIdentity1,
+			labelIdentityResImp1,
+			resImpNamespacedName,
+			&mcsv1alpha1.LabelIdentitySpec{},
+			true,
+		},
+	}
+
+	for _, tt := range tests {
+		fakeClient := fake.NewClientBuilder().WithScheme(scheme).WithObjects(tt.existLabelIdentity).Build()
+		fakeRemoteClient := fake.NewClientBuilder().WithScheme(scheme).WithObjects(tt.existResImp).Build()
+		remoteCluster := NewFakeRemoteCommonArea(fakeRemoteClient, "leader-cluster", localClusterID, "default")
+		r := NewLabelIdentityResourceImportReconciler(fakeClient, scheme, fakeClient, localClusterID, "default", remoteCluster)
+		if tt.isDeleted {
+			r.remoteCommonArea.Delete(ctx, tt.existResImp)
+		}
+
+		resImpReq := ctrl.Request{NamespacedName: tt.resImportNamespacedName}
+		if _, err := r.Reconcile(ctx, resImpReq); err != nil {
+			t.Errorf("LabelIdentityResourceImport Reconciler should handle LabelIdentity event successfully but got error = %v", err)
+		} else {
+			actLabelIdentity := &mcsv1alpha1.LabelIdentity{}
+			err := fakeClient.Get(ctx, types.NamespacedName{Namespace: "", Name: tt.resImportNamespacedName.Name}, actLabelIdentity)
+			if err == nil {
+				if !reflect.DeepEqual(*tt.expLabelIdentity, actLabelIdentity.Spec) {
+					t.Errorf("LabelIdentityResourceImport Reconciler imported a LabelIdentity incorrectly. Exp: %v, Act: %v", *tt.expLabelIdentity, actLabelIdentity.Spec)
+				}
+			} else {
+				if tt.isDeleted {
+					if !apierrors.IsNotFound(err) {
+						t.Errorf("LabelIdentityResourceImport Reconciler expects not found error but got error = %v", err)
+					}
+				} else {
+					t.Errorf("Expected a LabelIdentity but got error = %v", err)
+				}
+			}
+		}
```

Comment:
1. Since it uses existing object in `fakeRemoteClient` to represent Create and Update scenarios, I think it should just not add the object to `fakeRemoteClient` to represent Delete scearios, or all of them should call r.remoteCommonArea's Create/Update/Delete methods. `isDeleted` should be not needed.
2. The verification part could be simpler, it just needs to check whether the actual error and object matches, and get rid of case specific check
```
resImpReq := ctrl.Request{NamespacedName: tt.resImportNamespacedName}
err := r.Reconcile(ctx, resImpReq)
require.NoError(t, err, "LabelIdentityResourceImport Reconciler should handle LabelIdentity event successfully but got error")
actLabelIdentity := &mcsv1alpha1.LabelIdentity{}
getErr = fakeClient.Get(ctx, types.NamespacedName{Namespace: "", Name: tt.resImportNamespacedName.Name}, actLabelIdentity)
assert.Equal(t, tt.expGetErr, getErr)
assert.Equal(t, tt.expLabelIdentity, actLabelIdentity)
```

Reply:
Fixed